### ISSUE TITLE
Do a csv output of the CoBL bibliographical references ('sources') table

### ISSUE THREAD

**Issue by PaulHeggarty** [lingdb/CoBL#195]
This would then be something that Konstantin could use to speed up his search for all those references and put them into a Zotero folder, from which we could then export them in JSON format back to you, for when we move to the better system.

**Comment by runjak** [comment]
I've used the postgresql native `COPY` command to obtain the csv:
`COPY lexicon_source TO '/tmp/lexicon_source.csv' DELIMITER ',' CSV HEADER;`
I had to rename it to `.txt` for the upload:
[lexicon_source.txt](https://github.com/lingdb/CoBL/files/295534/lexicon_source.txt)

**Comment by PaulHeggarty** [comment]
Thanks.  Looks like the crucial thing will be to keep the id field, and somehow put it into the Zotero export too, if possible.  Perhaps a tag would work, but I’m not sure if they get output in the JSON export that Zotero does.

**Comment by PaulHeggarty** [comment]
I’ve now changed the labels and assignments on this issue to reflect that Jakob's part is done for now, and now it's Konstantin's baby to create the Zotero database.  Konstantin, are you familiar with Zotero?  If not, I can show you what I mean when we can meet up next.

**Comment by runjak** [comment]
If the `id` cannot be preserved we can still try to match the new source data against the old later on. I could generate a comparison for this that could be verified again.
The Zotero data will likely also hint at improvements to the current layout of the source table.

**Comment by nitsnatsnok** [comment]
Beautiful! I already worked with Zotero before switching to biblatex/biber, which I'd rather use: As it works with plain text, the reimport as well as adding an id field should be quite easy, this saves us from dealing with json & friends. I'll add that to my task list.
We could even think about putting the biblatex file in the github repo and have it synchronized with the database. This would allow simple export in at least this format, conversion to other formats could be dealt with in a second phase. Zotero by the way supports the import of bib(la)tex files – I'll test that.

To summarize, with regard to bibliographical references in general we need to:
- create a proper input mask (#166 is related)
- adapt the existing sources into a proper bibliographic reference system (#195 and #148)
- create consistent output from the bibliography (#183)

**Comment by runjak** [comment]
* If you prefer biblatex/biber that is fine for me - I've worked with the formats before but not in an automated fashion - parsing JSON would be straight forward, but I expect there to be `.bib` parsers for python, too.
* If the sources are going to be kind of static I think it would be a good idea to add them to the repo. Perhaps a good place can be found in the static directory. If on the other hand the sources will change much in the future I'd prefer to store them in the database mainly.

**Comment by PaulHeggarty** [comment]
It doesn’t have to be zotero, obviously, although I do find that a superbly fast way of adding bibliography entries.  But so long as an alternative works with open source software it's ok.  Perhaps best would be maximum compatibility with other CLLD resources like WOLD and WALS bibliographical references.

**Comment by runjak** [comment]
From my knowledge Zotero mainly uses methods already used in other places as well.
That is for embedding in websites it uses [RDF](https://en.wikipedia.org/wiki/Resource_Description_Framework) which is a known standard to exchange information. The same holds for `.bib` files which are widely used with [LaTeX](https://en.wikipedia.org/wiki/LaTeX) when creating papers or other text works. [JSON](http://json.org/) would likely be the easiest for me to integrate into the database, but we should aim to provide all those formats of the same data to clients so that everyone can easily use our sources independent of how they prepare their documents. This of course means that it would also work with Zotero.
